crates/adapters/src/transport/kafka/ft/mod.rs

```diff
@@ -460,6 +467,33 @@ where
     fn log(&self, level: RDKafkaLogLevel, fac: &str, log_message: &str) {
         self.deferred_logging.log(level, fac, log_message);
     }
+
+    fn generate_oauth_token(&self, _: Option<&str>) -> Result<OAuthToken, Box<dyn Error>> {
+        // TODO: Currently, OAUTHBEARER only works with AWS MSK.
+        if self.oauthbearer {
+            let region = {
+                let region = std::env::var("AWS_REGION").ok();
+                let default = std::env::var("AWS_DEFAULT_REGION").ok();
+                aws_types::region::Region::new(
+                    region.or(default).unwrap_or("us-east-1".to_string()),
+                )
+            };
+            let (token, expiration_time_ms) =
+                { futures::executor::block_on(async { generate_auth_token(region).await }) }?;
+
+            return Ok(OAuthToken {
+                token,
+                principal_name: "".to_string(),
+                lifetime_ms: expiration_time_ms,
+            });
+        }
+
+        Ok(OAuthToken {
+            token: "".to_string(),
+            principal_name: "".to_string(),
+            lifetime_ms: i64::MAX,
+        })
+    }
```

Comment:
This looks like it duplicates a lot of code with another function, can they be factored into a common helper?